pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -274,6 +275,14 @@ func checkVolumeSatisfyClaim(volume *v1.PersistentVolume, claim *v1.PersistentVo
 		return fmt.Errorf("storageClassName does not match")
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.VolumeAttributesClass) {
```

Comment:
Another alternative is if the feature gate is disabled but PVC or PV object has VAC set, then we should not bind at all. The user will have to delete and recreate the objects without VAC if they want to disable the feature.

Reply:
@msau42 and I were talking offline. I agree with her. Since the only way to have unbound PV/PVC with the VAC feature gate disabled is by creating the objects when the cluster had the gate enabled, there's no concern with disrupting existing workloads.

In this case I think the least surprising and most predictable thing to do is not bind objects with VAC when the feature gate is disabled.

Reply:
done